### DISCUSSION TITLE
Customizing the task page by adding "labels" field.

### DISCUSSION THREAD

**Discussion by keshva23** [cvat-ai/cvat#3920] [Q&A] [answered]
In CVAT, we were able to add labels like car,bus etc in Project creation page. My query is , how can we customize to add the labels in the Task creation page ?

**Comment by azhavoro** [comment, marked as answer]
Tasks created in a project will have exactly project labels, you cannot use a different labeling configuration for the project task. If you need a different label configuration, you can create single task without a project.

**Reply by keshva23** [reply]
Thank you for the reply.   

Here we were able to create a task with labels added from task page . But when we open the created task, we are not getting the labels because the project_id field in the "engine_label" table is shown as null.

**Comment by azhavoro** [comment]
`/api/v1/tasks?id=<task_id>` endpoint returns full task specification which contains `labels` field with label definition. Have I missed smth?

**Reply by keshva23** [reply]
Correct, Just now i checked by inspecting in browser console. 

I got the labels field as empty.

**Reply by azhavoro** [reply]
It should be impossible to create a task without labels in any case, how can I reproduce this?

**Reply by keshva23** [reply]
I want to disable adding labels in Project level, and keep adding label in tasks level and list the same labels for the jobs under that particular task.  Right now the labels created under task are not getting displayed under its Jobs.

**Reply by azhavoro** [reply]
Ok, now it's clear. This is the expected behavior, as I said, all project tasks must have the same labels. The projects were designed to be interpreted as a dataset. So, it's possible to configure labels for all project tasks at the project level only.

**Reply by keshva23** [reply]
@azhavoro  We got it. Thank you for the reply.
